sufia-models/lib/sufia/models/stats/user_stat_importer.rb

```diff
@@ -38,8 +38,12 @@ def date_since_last_cache(user)
       end
     end
 
-    def files_for_user(user)
-      ::GenericFile.where(Solrizer.solr_name('depositor', :symbol) => user.user_key)
+    def file_ids_for_user(user)
+      ids = []
+      ::GenericFile.find_in_batches("#{Solrizer.solr_name('depositor', :symbol)}:\"#{user.user_key}\"",{fl:["id"]}) do |group|
```

Comment:
Just one more thing, I promise. You can do `fl: 'id'` there's no need to wrap that in an array. Then, you can also remove the curly braces since there's only one hash parameter to find_in_batches.